src/sql-server-util/src/inspect.rs

```diff
@@ -110,6 +111,99 @@ pub fn get_changes_asc(
     )
 }
 
+/// Cleans up the change table associated with the specified `capture_instance` by
+/// deleting all entries with a `start_lsn` less than `low_water_mark`.
+///
+/// See: <https://learn.microsoft.com/en-us/sql/relational-databases/system-stored-procedures/sys-sp-cdc-cleanup-change-table-transact-sql?view=sql-server-ver16>.
+pub async fn cleanup_change_table(
+    client: &mut Client,
+    capture_instance: &str,
+    low_water_mark: &Lsn,
+    max_deletes: u32,
+) -> Result<(), SqlServerError> {
+    static GET_LSN_QUERY: &str =
+        "SELECT MAX(start_lsn) FROM cdc.lsn_time_mapping WHERE start_lsn <= @P1";
+    static CLEANUP_QUERY: &str = "
+DECLARE @mz_cleanup_status_bit BIT;
+SET @mz_cleanup_status_bit = 0;
+EXEC sys.sp_cdc_cleanup_change_table
+    @capture_instance = @P1,
+    @low_water_mark = @P2,
+    @threshold = @P3,
+    @fCleanupFailed = @mz_cleanup_status_bit OUTPUT;
+SELECT @mz_cleanup_status_bit;
+    ";
+
+    let max_deletes = i64::cast_from(max_deletes);
```

Comment:
What is this limit for? I saw that SQL Server has a parameter that defaults to 5000 but as far as our source is concerned we always want to cleanup everything otherwise we risk accumulating garbage with a high throughput enough table, right?

If we want to keep this parameter with a low value to control batching, and e.g avoid a single cleanup call taking a long time, then this function should be running the cleanup function iteratively until there are no more records to cleanup.

Reply:
My understanding is that this limit is not per LSN but per invocation of the cleanup function, right? So if we don't call it often enough we will fall behind. Since we commit to persist once per second the best case scenario is cleaning up 5000 records per second.

Until we make this an iterative process I would increase the default value of the parameter to something much much higher, like 1 million.

Reply:
Updated to 1,000,000!